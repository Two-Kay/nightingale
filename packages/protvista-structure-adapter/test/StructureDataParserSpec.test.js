import {
  getAllFeatureStructures,
  mergeOverlappingIntervals,
} from "../src/structure-data-parser";
import entryStructureData from "./UniProtEntryData";

describe("StructureDataParser", () => {
  it("should turn structures into features", () => {
    const features = getAllFeatureStructures(entryStructureData);
    expect(features).toMatchSnapshot();
  });

  it("should merge Overlapping Intervals", () => {
    const features = getAllFeatureStructures(entryStructureData);
    const overlapping = mergeOverlappingIntervals(features);
    expect(overlapping).toMatchSnapshot();
  });
});
