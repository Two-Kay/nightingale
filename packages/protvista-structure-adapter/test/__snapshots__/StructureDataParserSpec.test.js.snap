// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StructureDataParser should merge Overlapping Intervals 1`] = `
Array [
  Object {
    "accession": "P01234",
    "category": "STRUCTURE_COVERAGE",
    "data": Object {
      "Structures": Array [
        Object {
          "description": "Method: X-ray. Resolution: 1.50 A. ",
          "end": 0,
          "source": Object {
            "id": "1AAP",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AAP",
          },
          "start": 0,
        },
      ],
    },
    "end": 0,
    "start": 0,
    "type": "PDBE_COVER",
  },
  Object {
    "accession": "P01234",
    "category": "STRUCTURE_COVERAGE",
    "data": Object {
      "Structures": Array [
        Object {
          "description": "Method: NMR. Chains: A=672-699. ",
          "end": 699,
          "source": Object {
            "id": "1AMB",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AMB",
          },
          "start": 672,
        },
        Object {
          "description": "Method: NMR. Chains: A=672-699. ",
          "end": 699,
          "source": Object {
            "id": "1AMC",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AMC",
          },
          "start": 672,
        },
        Object {
          "description": "Method: NMR. Chains: A=672-711. ",
          "end": 711,
          "source": Object {
            "id": "1AML",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AML",
          },
          "start": 672,
        },
        Object {
          "description": "Method: NMR. Chains: A=672-711. ",
          "end": 711,
          "source": Object {
            "id": "1BA4",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1BA4",
          },
          "start": 672,
        },
      ],
    },
    "end": 711,
    "start": 672,
    "type": "PDBE_COVER",
  },
]
`;

exports[`StructureDataParser should turn structures into features 1`] = `
Array [
  Object {
    "accession": "P01234",
    "category": "STRUCTURE_COVERAGE",
    "data": Object {
      "Structures": Array [
        Object {
          "description": "Method: X-ray. Resolution: 1.50 A. ",
          "end": 0,
          "source": Object {
            "id": "1AAP",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AAP",
          },
          "start": 0,
        },
      ],
    },
    "end": 0,
    "start": 0,
    "type": "PDBE_COVER",
  },
  Object {
    "accession": "P01234",
    "category": "STRUCTURE_COVERAGE",
    "data": Object {
      "Structures": Array [
        Object {
          "description": "Method: NMR. Chains: A=672-699. ",
          "end": 699,
          "source": Object {
            "id": "1AMB",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AMB",
          },
          "start": 672,
        },
      ],
    },
    "end": 699,
    "start": 672,
    "type": "PDBE_COVER",
  },
  Object {
    "accession": "P01234",
    "category": "STRUCTURE_COVERAGE",
    "data": Object {
      "Structures": Array [
        Object {
          "description": "Method: NMR. Chains: A=672-699. ",
          "end": 699,
          "source": Object {
            "id": "1AMC",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AMC",
          },
          "start": 672,
        },
      ],
    },
    "end": 699,
    "start": 672,
    "type": "PDBE_COVER",
  },
  Object {
    "accession": "P01234",
    "category": "STRUCTURE_COVERAGE",
    "data": Object {
      "Structures": Array [
        Object {
          "description": "Method: NMR. Chains: A=672-711. ",
          "end": 711,
          "source": Object {
            "id": "1AML",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1AML",
          },
          "start": 672,
        },
      ],
    },
    "end": 711,
    "start": 672,
    "type": "PDBE_COVER",
  },
  Object {
    "accession": "P01234",
    "category": "STRUCTURE_COVERAGE",
    "data": Object {
      "Structures": Array [
        Object {
          "description": "Method: NMR. Chains: A=672-711. ",
          "end": 711,
          "source": Object {
            "id": "1BA4",
            "url": "http://www.ebi.ac.uk/pdbe-srv/view/entry/1BA4",
          },
          "start": 672,
        },
      ],
    },
    "end": 711,
    "start": 672,
    "type": "PDBE_COVER",
  },
]
`;
